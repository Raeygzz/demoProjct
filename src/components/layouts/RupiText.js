import React, { Component } from 'react';
import { Text } from 'react-native';


export default class RupiText extends Component {
  render() {
    return (
      <Text style={[{ fontFamily: "Apercu_Regular", color: "#fff", fontSize: 16 }, this.props.style]} minimumFontScale={16} allowFontScaling={this.props.allowFontScaling} adjustsFontSizeToFit={this.props.adjustsFontSizeToFit}>
        {this.props.text}
      </Text>
    );
  }
}


export class RupiBlackText extends Component {
  render() {
    return (
      <Text style={[{ fontFamily: "Apercu_Regular", color: "#223741", fontSize: 16 }, this.props.style]} minimumFontScale={16} allowFontScaling={this.props.allowFontScaling} adjustsFontSizeToFit={this.props.adjustsFontSizeToFit}>
        {this.props.text}
        {this.props.children}
      </Text>
    );
  }
}
