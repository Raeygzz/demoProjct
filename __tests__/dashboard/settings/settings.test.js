import React from 'react';
import 'react-native';

import Settings from '../../../src/screens/dashboard/settings/settings';
import userDemoImage from '../../../src/components/assets/images/demoUser.jpeg';
import RupiText from '../../../src/components/layouts/RupiText';
import globalStyle from '../../../src/components/styles/styles'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


import Enzyme from 'enzyme';
import { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


// const navigation = {
//   navigate: jest.fn()
// }


describe('<Settings />', () => {
  
  it('renders correctly', () => {
    const tree = renderer.create(<Settings />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders ImageBackground, TouchableOpacity, StyleSheet, globalStyle, TextInput, View, ScrollView, NormalButton, InvertButton, BaseGradient, FalseHeader, RupiCheckbox & RupiText components using (enzyme)expect, (enzyme)find, (enzyme)exists and (jest)toEqual', () => {
    const wrapper = shallow(<Settings />);
    expect(wrapper.find('View').exists()).toEqual(true);
    expect(wrapper.find('ImageBackground').exists()).toEqual(true);
    expect(wrapper.find('TouchableOpacity').exists()).toEqual(true);
    expect(wrapper.find('StyleSheet')).toBeTruthy();
    expect(wrapper.find('ScrollView')).toBeTruthy();
    expect(wrapper.find('TextInput').exists()).toEqual(true);
    expect(wrapper.find('globalStyle')).toBeDefined();
    expect(wrapper.find('RupiText').exists()).toEqual(true);
    expect(wrapper.find('BaseGradient').exists()).toEqual(true);
    expect(wrapper.find('FalseHeader').exists()).toEqual(true);
    expect(wrapper.find('RupiCheckbox').exists()).toEqual(true);
    expect(wrapper.find('NormalButton').exists()).toEqual(true);
    expect(wrapper.find('InvertButton').exists()).toEqual(true);
  });


  it('test state and state value using (enzyme)shallow,state,expect & (jest)toBeTruthy,toBeDefined,toBe', () => {
    const wrapper = shallow(<Settings />);
    expect(wrapper.state()).toBeTruthy();
    expect(wrapper.state()).toBeDefined();
    expect(wrapper.state('userProfileImage')).toBe('');
    expect(wrapper.instance().state.userFullName).toBe('');
  })

  it('test image component using (enzyme)shallow,expect,find & (jest)toBeTruthy,toEqual', () => {
    const wrapper = shallow(<Settings />);
    expect(wrapper).toBeTruthy();
    expect(wrapper.find('ImageBackground').length).toEqual(1);
    expect(wrapper.find("ImageBackground").prop("source")).toEqual(userDemoImage);
  })


  // it('renders the RupiText text User Name using (enzyme)shallow,expect,contains & (jest)toEqual', () => {
  //   const wrapper = shallow(<Settings />);
  //   expect(wrapper.length).toEqual(1);
  //   expect(wrapper.contains(<RupiText text="Regan Timsina" style={{fontSize: 36, lineHeight: 45, fontFamily: "Apercu Bold", color: "#223741", flex: 1 }}></RupiText>)).toEqual(true);
  // });


  // it('renders the title with enzyme', () => {   //commented on nov-29 enzyme utility(PASSED), as because expect property is invalid in chai framework
  //   const wrapper = shallow(<Settings title="Regan Timsina" />);
  //   expect(wrapper.find("RupiText").render().text()).toEqual("Regan Timsina");
  // });


  // it('renders the title with enzyme', () => {   //commented on nov-29 enzyme utility(PASSED), as because expect property is invalid in chai framework
  //   const wrapper = shallow(<Settings title="Regan Timsina" />);
  //   // expect(wrapper.text()).toEqual("Regan Timsina");
  //   expect(wrapper.children()).toEqual("Regan Timsina");
  // });


  // it('renders the RupiText text USER THEME using (enzyme)shallow,expect,contains & (jest)toEqual', () => {
  //   const wrapper = shallow(<Settings />);
  //   expect(wrapper.length).toEqual(1);
  //   expect(wrapper.contains(<RupiText text="SELECT THEME" style={{ fontSize: 16, lineHeight: 29, marginVertical: 10, color: "#223741", opacity: 0.7 }} ></RupiText>)).toEqual(true);
  // });


  // it('renders the RupiText text USER PERSONAL NOTE using (enzyme)shallow,expect,contains & (jest)toEqual', () => {
  //   const wrapper = shallow(<Settings />);
  //   expect(wrapper.length).toEqual(1);
  //   expect(wrapper.contains(<RupiText text="USER PERSONAL NOTE" style={[globalStyle.formHeadingNoMarBot, { marginBottom: 20 }]}></RupiText>)).toEqual(true);
  // });

})
