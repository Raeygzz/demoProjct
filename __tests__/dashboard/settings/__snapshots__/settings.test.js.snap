// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 39,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#10354b",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": -1,
        },
        undefined,
      ]
    }
  >
    <BVLinearGradient
      angle={135}
      colors={
        Array [
          4288230089,
          4283570823,
          4279121480,
          4280039002,
        ]
      }
      endPoint={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      locations={
        Array [
          0,
          0.36087942,
          0.8752376,
          1,
        ]
      }
      startPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 1534,
          "opacity": 0.32,
          "position": "relative",
          "zIndex": -1,
        }
      }
      useAngle={true}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 44,
          "justifyContent": "space-between",
          "marginBottom": 20,
          "marginTop": 55,
          "paddingHorizontal": 22,
        },
        undefined,
      ]
    }
  >
    <Text
      minimumFontScale={16}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Apercu_Regular",
            "fontSize": 16,
          },
          Array [
            Object {
              "fontSize": 20,
              "lineHeight": 25,
              "marginBottom": 24,
              "opacity": 0.69,
            },
            Object {
              "marginBottom": 0,
              "opacity": 1,
            },
          ],
        ]
      }
    >
      Settings
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 20,
              },
              Object {
                "color": "white",
              },
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 20,
              },
              Object {
                "color": "white",
                "marginLeft": 20,
              },
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ÓÅ•
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingTop": 10,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "marginTop": 26,
            },
            Object {
              "paddingBottom": 34,
              "paddingHorizontal": 27,
              "paddingTop": 50,
            },
            Object {
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
            Object {
              "paddingBottom": 61,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 9,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "height": 80,
                    "justifyContent": "center",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": -86,
                    "width": 80,
                  },
                  Object {
                    "borderRadius": 1000,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/components/assets/images/demoUser.jpeg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
          <Text
            minimumFontScale={16}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 16,
                },
                Object {
                  "color": "#223741",
                  "flex": 1,
                  "fontFamily": "Apercu Bold",
                  "fontSize": 36,
                  "lineHeight": 45,
                },
              ]
            }
          >
            Regan Timsina
          </Text>
        </View>
        <Text
          minimumFontScale={16}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Apercu_Regular",
                "fontSize": 16,
              },
              Object {
                "color": "#223741",
                "fontSize": 16,
                "lineHeight": 29,
                "marginVertical": 10,
                "opacity": 0.7,
              },
            ]
          }
        >
          SELECT THEME
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "marginBottom": 20,
            }
          }
        >
          <View
            style={
              Object {
                "marginVertical": 12,
                "paddingLeft": 40,
                "position": "relative",
              }
            }
          >
            <Text
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "bottom": 0,
                  "color": "transparent",
                  "fontSize": 0,
                  "left": 0,
                  "opacity": 0.5,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 9,
                }
              }
            >
              .
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 26,
                    "justifyContent": "center",
                    "left": 0,
                    "marginRight": 20,
                    "position": "absolute",
                    "top": 0,
                    "width": 26,
                    "zIndex": 0,
                  },
                  null,
                ]
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "Apercu Light",
                }
              }
            >
              Dark Theme
            </Text>
          </View>
          <View
            style={
              Object {
                "marginVertical": 12,
                "paddingLeft": 40,
                "position": "relative",
              }
            }
          >
            <Text
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "bottom": 0,
                  "color": "transparent",
                  "fontSize": 0,
                  "left": 0,
                  "opacity": 0.5,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 9,
                }
              }
            >
              .
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 26,
                    "justifyContent": "center",
                    "left": 0,
                    "marginRight": 20,
                    "position": "absolute",
                    "top": 0,
                    "width": 26,
                    "zIndex": 0,
                  },
                  null,
                ]
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "Apercu Light",
                }
              }
            >
              Light Theme
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffbcbc",
                  "borderColor": "red",
                  "borderWidth": 1,
                  "height": 100,
                  "justifyContent": "center",
                  "margin": 10,
                  "width": 300,
                },
                Object {
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#333333",
                  "margin": 20,
                }
              }
            >
              TouchableNativeFeedback is not supported on this platform!
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffbcbc",
                  "borderColor": "red",
                  "borderWidth": 1,
                  "height": 100,
                  "justifyContent": "center",
                  "margin": 10,
                  "width": 300,
                },
                Object {
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#333333",
                  "margin": 20,
                }
              }
            >
              TouchableNativeFeedback is not supported on this platform!
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#171635",
            },
            Object {
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
            Object {
              "marginTop": -37,
              "paddingBottom": 34,
              "paddingHorizontal": 20,
              "paddingTop": 30,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              minimumFontScale={16}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Apercu_Regular",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontFamily": "Apercu-Bold",
                      "fontSize": 14,
                      "lineHeight": 17,
                      "opacity": 0.48,
                    },
                    Object {
                      "marginBottom": 20,
                    },
                  ],
                ]
              }
            >
              USER PERSONAL NOTE
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "center",
              "paddingBottom": 75,
            }
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "darkgrey",
                  "borderColor": "#FFCA2D",
                  "borderWidth": 2,
                  "color": "#FFCA2D",
                  "fontSize": 16,
                  "height": 320,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Object {
                "right": -280,
                "top": 5,
                "width": "20%",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ffbcbc",
                    "borderColor": "red",
                    "borderWidth": 1,
                    "height": 100,
                    "justifyContent": "center",
                    "margin": 10,
                    "width": 300,
                  },
                  Object {
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333333",
                    "margin": 20,
                  }
                }
              >
                TouchableNativeFeedback is not supported on this platform!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
